### PR TITLE
Bug 1675625 - Refactor error handing in Mojolicious to allow for native mojo code to something other than Bugzilla::Error

### PR DESCRIPTION
**Pull request by dklawren**

### CODE REVIEW

File: Bugzilla/App/API.pm [hunk 1]
```diff
@@ -24,13 +27,27 @@ sub setup_routes {
   push @$namespaces, 'Bugzilla::API';
   $r->namespaces($namespaces);
 
+  # Backwards compat with /api/user/profile which Phabricator
```
Comment (globau) [inline Bugzilla/App/API.pm:30]: incomplete comment

File: Bugzilla/API/V1/User.pm [hunk 2]
```diff
@@ -13,8 +13,7 @@ use Mojo::JSON qw( true false );
 
 sub setup_routes {
   my ($class, $r) = @_;
-  $r->get('/api/user/profile')->to('V1::User#user_profile');
-  $r->get('/rest/user/profile')->to('V1::User#user_profile');
+  $r->get('/profile')->to('V1::User#user_profile');
```
Comment (globau) [inline Bugzilla/API/V1/User.pm:16]: Won't this end up being `/rest/profile` instead of `/rest/user/profile`?

**Review by globau — CHANGES_REQUESTED** [review]
You'll also need to update the routes in `Bugzilla/API/V1/Teams.pm`.

File: Bugzilla/API/V1/User.pm (hunk 2 above)
Comment (dklawren) [reply Bugzilla/API/V1/User.pm]: Yeah this is tricky. Having this as /rest/user/profile conflicts with Bugzilla::WebService::User::get in that the older code thinks 'profile' is a user name to return data for and instead errors that 'profile' cannot be found. With #1657 I will be able to properly fix this but for now I have renamed the endpoint to '/rest/user_profile'.

**Review by globau — APPROVED** [review]

**Pull request merged**